pkg/s3/bucket.go

```diff
@@ -146,22 +151,81 @@ func CreateBucketTaggingInput(bucketname string, tags map[string]string) *s3.Put
 	return putInput
 }
 
+// ClearBucketTags wipes all existing tags from a bucket so that velero-specific
+// tags can be applied to the bucket instead.
 func ClearBucketTags(s3Client *s3.S3, bucketName string) (err error) {
 	deleteInput := &s3.DeleteBucketTaggingInput{Bucket: aws.String(bucketName)}
 	_, err = s3Client.DeleteBucketTagging(deleteInput)
 	return err
 }
 
-func TagBucket(s3Client *s3.S3, bucketName string, backUpLocation string) error {
+// TagBucket adds tags to an S3 bucket. The tags are used to indicate that velero backups
+// are stored in the bucket, and to identify the associated cluster.
+func TagBucket(s3Client *s3.S3, bucketName string, backUpLocation string, infraName string) error {
 	err := ClearBucketTags(s3Client, bucketName)
 	if err != nil {
 		return fmt.Errorf("unable to clear %v bucket tags: %v", bucketName, err)
 	}
-	input := CreateBucketTaggingInput(bucketName, map[string]string{bucketTagKey: backUpLocation})
+	input := CreateBucketTaggingInput(bucketName, map[string]string{
+		bucketTagKey: backUpLocation,
+		"infraName":  infraName,
```

Comment:
`"infraName"` here should be a const at the top of the file. I'd also use a scoped name, like `velero.io/infrastructureName` as the tag key